verilog/analysis/checkers/dff_name_style_rule.cc

```diff
@@ -0,0 +1,288 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "verilog/analysis/checkers/dff_name_style_rule.h"
+
+#include <charconv>
+#include <set>
+#include <string>
+
+#include "absl/strings/match.h"
+#include "absl/strings/str_cat.h"
+#include "absl/strings/str_join.h"
+#include "absl/strings/str_split.h"
+#include "absl/strings/string_view.h"
+#include "common/analysis/lint_rule_status.h"
+#include "common/analysis/matcher/bound_symbol_manager.h"
+#include "common/analysis/matcher/matcher.h"
+#include "common/strings/naming_utils.h"
+#include "common/text/config_utils.h"
+#include "common/text/symbol.h"
+#include "common/text/syntax_tree_context.h"
+#include "common/text/token_info.h"
+#include "dff_name_style_rule.h"
+#include "verilog/CST/statement.h"
+#include "verilog/CST/verilog_matchers.h"
+#include "verilog/analysis/descriptions.h"
+#include "verilog/analysis/lint_rule_registry.h"
+#include "verilog/parser/verilog_token_enum.h"
+
+namespace verilog {
+namespace analysis {
+
+using verible::LintRuleStatus;
+using verible::LintViolation;
+using verible::SyntaxTreeContext;
+using verible::config::SetString;
+using verible::matcher::Matcher;
+
+// Register DffNameStyleRule.
+VERILOG_REGISTER_LINT_RULE(DffNameStyleRule);
+
+const LintRuleDescriptor &DffNameStyleRule::GetDescriptor() {
+  static const LintRuleDescriptor d{
+      .name = "dff-name-style",
+      .topic = "dff-name-style",
+      .desc =
+          "Checks that D Flip-Flops use appropiate naming conventions in both "
+          "input and output ports.",
+      .param = {{"input", kDefaultInputSuffixes,
+                 "Comma separated list of allowed suffixes for the input port. "
+                 "Suffixes should not include the preceding \"_\". Empty field "
+                 "means no checks for the input port."},
+                {"output", kDefaultOutputSuffixes,
+                 "Comma separated list of allowed suffixes for the input port. "
+                 "Should "
+                 "not include the preceding \"_\". Empty field means no checks "
+                 "for the output port."}}};
+  return d;
+}
+
+static const Matcher &AlwaysFFMatcher() {
+  static const Matcher matcher(NodekAlwaysStatement(AlwaysFFKeyword()));
+  return matcher;
+}
+
+void DffNameStyleRule::HandleSymbol(const verible::Symbol &symbol,
+                                    const SyntaxTreeContext &context) {
+  verible::matcher::BoundSymbolManager manager;
+  if (!AlwaysFFMatcher().Matches(symbol, &manager) ||
+      (valid_input_suffixes.empty() && valid_output_suffixes.empty())) {
+    return;
+  }
+
+  // Once we encounter an always_ff statement, extract inner non
+  // blocking assignments.
+  std::vector<verible::TreeSearchMatch> non_blocking_assignments =
+      FindAllNonBlockingAssignments(symbol);
+
+  for (auto &tree_match : non_blocking_assignments) {
```

Comment:
I think this can be `const auto&`, as match objects aren't being modified.

Reply:
Also, expand the type: make it `for (const verible::TreeSearchMatch &tree_match ...)` to reduce the use of `auto` and [keep it explicit for the benefit of the reader](https://google.github.io/styleguide/cppguide.html#Type_deduction).